Add service to glossary (#3815)
open-telemetry/opentelemetry-specification: specification/glossary.md

An umbrella term for the smallest unit of sequential code execution, used in different concepts of multitasking. Examples are threads, coroutines or fibers.

### Service

> OpenTelemetry has adopted a broad interpretation such that every telemetry source is a service. Examples include, but are not limited to: web services, hosts, mobile applications, browser application, edge computing devices, functions as a service, databases, message brokers, etc. See the [semantic conventions for `service`](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#service) for more details.

Reviewer: I don't mind this definition on its own, but it would be better to follow the established format in this doc (even though I personally don't like it)
```suggestion
A source of telemetry, broadly interpreted. Examples include, but are not limited to: web services, hosts, mobile applications, browser application, edge computing devices, functions as a service, databases, message brokers, etc. See the [semantic conventions for `service`](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#service) for more details.
```
